content/en/docs/reference/setup-tools/kubeadm/kubelet-integration.md

```diff
@@ -0,0 +1,88 @@
+## Configuring the kubelets in your cluster with kubeadm and packaging
+
+#### Introduction to the relation between kubeadm and kubelet
+
+The `kubeadm` CLI tool has a lifecycle that is decoupled from the lifecycle of the `kubelet` daemon, the [Kubernetes Node Agent](#TODO). The `kubeadm ` CLI tool is executed by the user at initialization or upgrade time. The kubelet is always running on any given master or node that is running Kubernetes. As the `kubelet` is a daemon, it needs to be run by some kind of init system or process manager. In the deb and rpm packages that are shipped with Kubernetes releases, the `kubelet` is by default run as a `systemd` service. _Important to note_ though, is that _any type_ of init system or process manager can be used for running the `kubelet`, this default solution is a reference solution, but swappable. By default, the `kubeadm` deb/rpm package ships a `systemd` dropin file for the kubelet, so it becomes functional. The vanilla `kubelet` deb/rpm package does not ship with any CLI flags at all (in other words, is non-functional).
+
+#### Problem 1: Cluster-level information might have to be propagated down to the kubelets
+
+There is a need for configuration values of different kinds to flow down from `kubeadm` that is run at `kubeadm init` and/or `kubeadm join` time. Common scenarios e.g. include customizing the Service Subnet and using an other CRI runtime. Let's say you pass this parameter to kubeadm:
+
+```
+kubeadm init --service-cidr 10.95.0.0/12
+```
+
+The default service subnet is `10.96.0.0/12` for reference.
+
+When in this example the subnet used for allocating Service Virtual IPs is modified, the `--cluster-dns` kubelet flag also needs to be customized, **for every kubelet in the cluster**. In this example, it should be set to `10.95.0.10`. This means we need some kind of way for global, cluster-specific configuration (e.g. Service Subnet) to "flow down" to the kubelets in the cluster.
+
+Luckily, the kubelet has a versioned, structured API object that can be used for configuring most of the knobs in the kubelet, hereafter called **the kubelet's ComponentConfig**. The API types for the kubelet ComponentConfig can be found [here](#TODO). It lets the user specify e.g. the cluster DNS IP with the following configuration file:
```

Comment:
looking for one of these for the TODO links?

* https://github.com/kubernetes/kubernetes/blob/release-1.11/pkg/kubelet/apis/kubeletconfig/types.go#L45
* https://godoc.org/k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig#KubeletConfiguration